src/git/github-git/repo-tags.ts

```diff
@@ -0,0 +1,102 @@
+import { execWithOutput } from "../../utils";
+
+export interface GitTag {
+  name: string;
+  ref: string;
+}
+
+/**
+ * Pushes any missing tags to the remote first replacing those tags with lightweight versions.
+ * We replace annotated tags with lightweight ones because we cannot sign annotated tags, but
+ * lightweight tags pointing to signed commits will show up as verified in GitHub.
+ */
+export async function pushTags(cwd?: string) {
+  const onlyLocalTags = await getOnlyLocalTags(cwd);
+
+  await deleteTags(onlyLocalTags, cwd);
+  const createdTags = await createLightweightTags(onlyLocalTags, cwd);
+  await execWithOutput("git", ["push", "origin", "--tags"], { cwd });
+
+  return createdTags;
+}
+
+export async function getOnlyLocalTags(cwd?: string): Promise<GitTag[]> {
+  const localTags = await getLocalTags(cwd);
```

Comment:
`getLocalTags` vs `getOnlyLocalTags` is kind of confusing. This is needed because the local repo will contain many remote tags that were obtained during the checkout process?

Might want to call out this distinction explicitly in a comment.